python-frontend/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -1967,6 +1972,20 @@ public void string_interpolation() {
     assertThat(elmt.isInterpolated()).isTrue();
     assertThat(elmt.formattedExpressions()).hasSize(1);
     assertThat(elmt.formattedExpressions().get(0).expression().is(Kind.SET_COMPREHENSION)).isTrue();
+
+    stringLiteral = (StringLiteral) parse("f\"{x if p != 0 else 0}\"", treeMaker::expression);
+    assertThat(stringLiteral.stringElements()).hasSize(1);
+    elmt = stringLiteral.stringElements().get(0);
+    assertThat(elmt.isInterpolated()).isTrue();
+    assertThat(elmt.formattedExpressions()).hasSize(1);
+    assertThat(elmt.formattedExpressions().get(0).expression().is(Kind.CONDITIONAL_EXPR)).isTrue();
+
+    stringLiteral = (StringLiteral) parse("f\"missing {'previous C' if pCatom is None else 'N'} atom\"", treeMaker::expression);
+    assertThat(stringLiteral.stringElements()).hasSize(1);
+    elmt = stringLiteral.stringElements().get(0);
+    assertThat(elmt.isInterpolated()).isTrue();
+    assertThat(elmt.formattedExpressions()).hasSize(1);
+    assertThat(elmt.formattedExpressions().get(0).expression().is(Kind.CONDITIONAL_EXPR)).isTrue();
```

Comment:
I don't see what those test cases bring. Maybe I'm missing something.

Reply:
I wanted to have a test ensuring the entire string gets parsed correctly, but indeed having 1 of them is debatable and 2 redundant, so I'll just keep one.